lib/vsteam-lib/InternalCache.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Management.Automation;
+using System.Management.Automation.Abstractions;
+
+namespace vsteam_lib
+{
+   public class InternalCache
+   {
+      private double _timeStamp = -1;
+      public List<string> Values { get; } = new List<string>();
+
+      /// <summary>
+      /// Used for unit testing
+      /// </summary>
+      private IPowerShell _powerShell;
+      private readonly string _command;
+      private readonly string _property;
+      private readonly bool _requiresProject;
+
+      public InternalCache(string command, string property, bool requiresProject)
+      {
+         this._command = command;
+         this._property = property;
+         this._requiresProject = requiresProject;
+      }
+
+      [ExcludeFromCodeCoverage]
+      public IPowerShell Shell
+      {
+         get
+         {
+            if (this._powerShell == null)
+            {
+               this._powerShell = new PowerShellWrapper(RunspaceMode.CurrentRunspace);
+            }
+
+            return this._powerShell;
+         }
+
+         set => this._powerShell = value;
+      }
+
+      internal void Update(IEnumerable<string> list)
+      {
+         // If a list is passed in use it. If not call Get-VSTeamProcess
+         if (null == list)
+         {
+            if (this._requiresProject)
+            {
+               var projectName = Common.GetDefaultProject(this.Shell);
+
+               if (!string.IsNullOrEmpty(projectName))
+               {
+                  this.Shell.Commands.Clear();
+
+                  list = this.Shell.AddCommand(this._command)
+                                   .AddParameter("ProjectName", projectName)
+                                   .AddCommand("Select-Object")
+                                   .AddParameter("ExpandProperty", this._property)
+                                   .AddCommand("Sort-Object")
+                                   .Invoke<string>();
+               }
+            }
+            else
+            {
+               this.Shell.Commands.Clear();
+
+               list = this.Shell.AddCommand(this._command)
+                                .AddCommand("Select-Object")
+                                .AddParameter("ExpandProperty", this._property)
+                                .AddCommand("Sort-Object")
+                                .Invoke<string>();            
+            }
+         }
+
+         this.Prime(list);
+      }
+
+      internal void Prime(IEnumerable<string> list)
```

Comment:
What is it for? Why is it named "Prime"?
Update: I can see that it is clearing a cache variable and repopulating it, but I don't know what the wording has to do with the business logic.

Reply:
This is for priming the cache with a list. For example, if you call Get-VSTeamProject you can prime the cache with the projects you just got back. You prime pumps by filling them with water before you use them. We are just priming the cache before we use it.

But we can change it. What word makes the purpose more clear?

Reply:
something more direct like PreFill or Initialize?

Reply:
I will change to PreFill